emsdk

```diff
@@ -1697,10 +1707,13 @@ def download_waterfall_lkgr():
 
 def load_waterfall_lkgr():
   try:
-    data = json.loads(open(os.path.join('upstream', 'lkgr.json'), 'r').read())
+    text = open(sdk_path(os.path.join('upstream', 'lkgr.json')), 'r').read()
+    data = json.loads(text)
     lkgr = data['build']
     return [lkgr]
-  except:
+  except Exception as e:
+    print('Error parsing lkgr.json!')
```

Comment:
Yay better debug messages

Although I think what I was thinking here was that a user may not have fetched the waterfall lkgr ever, say they upgraded emsdk from before it was included, and I think this code gets hit when invoking `emsdk list`?

Anyway this is harmless enough, and in the future will be the default, so this is still a good change. Was trying to dump my context of "why did I ever make this fail silently?"

Reply:
btw, what is lkgr short of?

Reply:
Hmm indeed it looks like `upstream/lkgr.json` might not exist? Perhaps `if not os.path.isfile(os.path.join('upstream', 'lkgr.json'): return []` ? (There is a `VERBOSE` variable top of the file if you want to do some logging only in a verbose/debug mode in this case)

Reply:
I think it's "last known good revision".

Yes, it looks like the file might not exist if tags have not been updated. We could print "you should run update-tags" in that case, however, looks like that would be printed during the startup of the emsdk invocation that you run update-tags with, unless we added some more logic, not sure offhand how simple that would be.